Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -302,9 +302,12 @@ void MemoryWidget::ConnectWidgets()
 
   connect(m_input_combo, qOverload<int>(&QComboBox::currentIndexChanged), this,
           &MemoryWidget::ValidateAndPreviewInputValue);
-
-  connect(m_set_value, &QPushButton::clicked, this, &MemoryWidget::OnSetValue);
-  connect(m_from_file, &QPushButton::clicked, this, &MemoryWidget::OnSetValueFromFile);
+  connect(m_set_value, &QPushButton::clicked, [this]() {
+    if (m_input_combo->currentData().toInt() == -1)
+      OnSetValueFromFile();
+    else
+      OnSetValue();
+  });
```

Comment:
`this, [this]() {` should be used as it will ensure that:
 - the signal will be executed only if `this` still exists
 - the lambda will be disconnected if the sender or the receiver is destroyed

Reply:
Ah thanks for the clarification!